go/multipooler/manager/rpc_backup.go

```diff
@@ -122,22 +122,38 @@ func (pm *MultiPoolerManager) RestoreFromBackup(ctx context.Context, backupID st
 	poolerDir := filepath.Dir(configPath)
 	pgDataDir := filepath.Join(poolerDir, "pg_data")
 
-	// Determine if we should maintain standby status after restore
-	// We query PostgreSQL directly to get the current recovery status
-	slog.InfoContext(ctx, "Checking recovery status before restore")
-	isPrimary, err := pm.isPrimary(ctx)
-	if err != nil {
-		slog.ErrorContext(ctx, "Failed to check recovery status before restore", "error", err)
-		return mterrors.Wrap(err, "failed to check recovery status")
-	}
+	// Check if PostgreSQL is initialized by looking for PG_VERSION file
+	pgVersionPath := filepath.Join(pgDataDir, "PG_VERSION")
+	_, err := os.Stat(pgVersionPath)
+	uninitialized := os.IsNotExist(err)
```

Comment:
We also have a definition of whether the pooler's postgres has been initialized as
```
func (pm *MultiPoolerManager) isInitialized() bool {
```
and maybe there are other places where we are duplicating similar logic. We should consolidate them all in one place and do it one way.